### PR TITLE
[RHOAIENG-8124] Show Pipeline Group Label on Hover at Low Scale

### PR DESCRIPTION
**Pull request by jenny-s51**
### Closes https://issues.redhat.com/browse/RHOAIENG-8124:

Before
Screenshot 2024-06-20 at 11 52 41 AM

After
Screenshot 2024-06-20 at 11 36 30 AM

### Also closes https://issues.redhat.com/browse/RHOAIENG-8370:

Before
before

After
Screenshot 2024-06-20 at 1 22 37 PM

## Description

Updates pipelines group to show label when hovered to provide access to expand/collapse actions.

## How Has This Been Tested?

To test label updates:

1. Navigate/upload pipelines example with groups: 
[flip_coin_compiled.yml.zip](https://github.com/user-attachments/files/15917833/flip_coin_compiled.yml.zip)

2. Expand a group
3. Zoom out to low scale
4. Verify that the label is shown when hovered and expand/collapse actions can be accessed in this UI state 

To test group boundary fix:

1. Upload [Pipeline time-series-dense-encoder-forecasting.yaml.zip](https://github.com/user-attachments/files/15917849/Pipeline.time-series-dense-encoder-forecasting.yaml.zip)

2. Verify that group boundaries persist when zooming in / out. 

## Test Impact

N/A

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/artifacts/ArtifactNodeDetails.tsx [hunk 1]
```diff
@@ -90,14 +88,7 @@ export const ArtifactNodeDetails: React.FC<ArtifactNodeDetailsProps> = ({
...
               <DescriptionListDescription>
-                <Stack hasGutter>
-                  <StackItem>
-                    <ArtifactUriLink uri={artifact.getUri()} />
-                  </StackItem>
-                  <StackItem>
-                    <ArtifactPreview artifact={artifact} />
-                  </StackItem>
-                </Stack>
+                <ArtifactUriLink uri={artifact.getUri()} />
```
Comment (jeff-phillips-18) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/artifacts/ArtifactNodeDetails.tsx:91]: I'm not following how these changes are relevant to the issues being addressed in this PR.

Comment (jenny-s51) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/artifacts/ArtifactNodeDetails.tsx]: Thank you @jeff-phillips-18 rebased

**Review by jeff-phillips-18 — APPROVED** [review]
/lgtm

**Review by DaoDaoNoCode — APPROVED** [review]
Tested, working great, thanks for the updates!
/lgtm

**Comment by DaoDaoNoCode** [comment]
@jenny-s51 Could you rebase the PR?

**Review by jenny-s51 — COMMENTED** [review]
Thank you @DaoDaoNoCode , rebased

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm

**Comment by Gkrumbach07** [comment]
looks good

/approve

**Pull request merged**